File: plugins/dotnet/robocode.dotnet.napi/src/net/sf/robocode/security/IThreadManagerBase.cs
using IKVM.Attributes;
using java.io;

namespace net.sf.robocode.security
{
    public interface IThreadManagerBase
    {
        [Throws(new[] {"java.io.IOException"})]
        FileOutputStream createRobotFileStream(string str, bool b);

        bool isSafeThread();
    }
}

File: robocode.api/src/main/java/net/sf/robocode/security/IThreadManagerBase.java
/**
 * Copyright (c) 2001-2019 Mathew A. Nelson and Robocode contributors
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * https://robocode.sourceforge.io/license/epl-v10.html
 */
package net.sf.robocode.security;


import java.io.FileOutputStream;
import java.io.IOException;


/**
 * @author Pavel Savara (original)
 */
public interface IThreadManagerBase {
	boolean isSafeThread();
	FileOutputStream createRobotFileStream(String fileName, boolean append) throws IOException;
}
